src/main/java/net/mcreator/blockly/java/blocks/MCItemBlock.java

```diff
@@ -39,7 +44,16 @@ public class MCItemBlock implements IBlockGenerator {
 						L10N.t("blockly.errors.mcitem_broken_reference",
 								textContent.replaceFirst(NameMapper.MCREATOR_PREFIX, ""))));
 			}
-			master.append(new NameMapper(master.getWorkspace(), "blocksitems").getMapping(textContent));
+
+			if (master.getTemplateGenerator() != null && master.getTemplateGenerator()
+					.hasTemplate("_mcitem.java.ftl") && block.getAttribute("type").equals("mcitem_all")) {
+				Map<String, Object> dataModel = new HashMap<>();
+				dataModel.put("block", new MItemBlock(master.getWorkspace(), textContent));

```

Comment:
Calling this block is not ideal since this can also be item

Reply:
Good catch, this is originally meant for blocks too. Left comment in your other PR too where this should be improved too for more clear data model: https://github.com/MCreator/MCreator/pull/5827/files#r2648339927

Reply:
Ok, good. No problem, I'll do the change